### PR TITLE
Added some stuff for metrics

### PR DESCRIPTION
**Pull request by calvertj**

### CODE REVIEW

File: scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala [hunk 1]
```diff
@@ -15,11 +15,36 @@
...
+/**
+ * Represents the basic instance types that can be requested
+ * such as aws c3.large
+ * @param name A unique name such as c3.large per vendor
+ * @param mem The total memory of a node
+ * @param cpu The cpu count of a node normalized to aws CPUs
+ * @param io The max sustained IOps of a node
+ * @param vendor The name of the vendor such as amazon aws
+ */
+case class NodeType(name: String, mem: Int, cpu: Int, io: Int, vendor: String)
```
Comment (tdanford) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala]: We should talk about design -- I don't want to go quite so detailed yet.  At most I might model memory first.

File: scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala [hunk 2]
```diff
@@ -29,7 +54,7 @@ case class Node(nodeId : String) {
  * Either startup a node, or shut an existing node down.
  */
 trait ElasticNodeResource {
-
-  def requestNode()
```
Comment (tdanford) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala:33]: I actually want to take a look at Akka to define the interfaces between ... a couple actor types that I've drawn out in my notebook but haven't described here yet.  BTW, J, have you seen this: https://github.com/aphyr/timelike ?

File: scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala (hunk 1 above)
Comment (calvertj) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala]: cool,
Send me a pick, ill make it in akka,

J

Comment (calvertj) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala]: Akka is awesome, so glad they rolled it in.  have you see the new akka
streams (CEP),

J

Comment (tdanford) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala]: J, drop the cpu, io, vendor params and I'll merge this :-)

File: scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala (hunk 2 above)
Comment (calvertj) [inline scheduling-core/src/main/scala/com/tdanford/simulator/Node.scala:33]: Akka done right means no dealing with threads ;)

**Comment by calvertj** [comment]
opening a new pull request...

**Comment by tdanford** [comment]
you know, you can just rebase the branch and it'll update the PR...

**Comment by calvertj** [comment]
I do now...

**Comment by calvertj** [comment]
you know, you could just give me write access so I can commit, don't worry,
its a versioning system

**Comment by tdanford** [comment]
I like accepting PRs :-)

**Pull request closed without merging**